src/main/java/com/github/exadmin/sourcesscanner/exclude/ExcludeFileModel.java

```diff
@@ -42,11 +42,9 @@ public void doSortBeforeSaving() {
     }
 
     public boolean isPathFullyIgnored(String relFileOrDirNameHash) {
-        for (ExcludeSignatureItem next : signatures) {
-            if (next.getFileHash().equals(relFileOrDirNameHash) && next.getTextHash().equals(SKIP_FULL_FILE_HASH)) return true;
-        }
+    return signatures.stream()
+        .anyMatch(item -> item.getFileHash().equals(relFileOrDirNameHash) 
```

Comment:
Consider extracting the complex predicate logic into a separate method for better readability.

```suggestion
        .anyMatch(this::isFullFileIgnored);
```

#deepseek-review:inline